neoexchange/core/models/blocks.py

```diff
@@ -228,10 +229,16 @@ def num_unique_red_frames(self):
     def num_spectro_frames(self):
         """Returns the numbers of different types of spectroscopic frames"""
         num_moltypes_string = 'No data'
-        data, num_frames = check_for_archive_images(self.request_number, obstype='')
+        try:
+            data, num_frames = check_for_archive_images(self.request_number, obstype='', obj=self.body.current_name())
```

Comment:
Does `self.current_name()` not work here to get either `Body` or `calibsource` name depending on which is defined (saving a try/except) ?

Reply:
that should work perfectly. good catch!